src/vis/vis.js

```diff
@@ -365,6 +349,14 @@ var VisModel = Backbone.Model.extend({
       includeFilters: true,
       success: function () {
         this.trigger('reloaded');
+        this._restartAnalysisPolling();
+
+        var analysisNodes = this._getAnalysisNodeModels();
+        if (this._isAnyAnalysisNodeLoading(analysisNodes)) {
+          this.trackLoadingObject(this);
```

Comment:
what does `trackLoadingObject` do?

Reply:
I mean, we keep a list of "loading objects" but we dont do anything with it :S

Reply:
It keeps track of "things" that are loading. I admit this is kind of weird now... There're two types of objects that might be "loading":

1. `vis` (when any analysis is still not done). This one is now tracked / untracked every time the vis is reloaded, based on the status of analyses.
2. `leafletCartoDBLayerGroupView` or `gmapsCartoDBLayerGroupView`. These will track or untrack themselves as loading events when tiles are loading / loaded.

We want to mark the vis as `loading` (to display the blue loader/bar at the top of the map) whenever there's any of those two objects are loading.

We're tracking each individual `AnalysisModel` before but I realised it doesn't make sense because as soon as one is not done, the vis is "loading" and the vis will be reloaded when an analysis pointed by a layer or a dataview is ready.